src/plugins/backend_odbc/backend_odbc_get.c

```diff
@@ -0,0 +1,827 @@
+	}
+	else if (bufferSize <= 0)
+	{
+		ELEKTRA_SET_INTERFACE_ERROR (errorKey, "A buffer size <=0 was given as an argument to getLongData()");
+		return false;
+	}
+
+	do
+	{
+		if (iteration > 0)
+		{
+			if (elektraRealloc ((void **) targetValue, bufferSize * (iteration + 1)) == -1)
+			{
+				SQLFreeHandle (SQL_HANDLE_STMT, sqlStmt);
+				ELEKTRA_SET_OUT_OF_MEMORY_ERROR (errorKey);
+				return false;
+			}
+		}
+
+		getDataRet = SQLGetData (sqlStmt, colNumber, targetType, (*targetValue) + (iteration * (bufferSize - 1)), bufferSize,
+					 &getDataLenOrInd);
+
+		if (SQL_SUCCEEDED (getDataRet))
+		{
+			if (getDataRet == SQL_SUCCESS_WITH_INFO)
+			{
+				setOdbcError (SQL_HANDLE_STMT, sqlStmt, "getLongData", true, errorKey);
+			}
+			else
+			{
+				/* The last call returns SQL_SUCCESS, no more iterations necessary
+				 * see: https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/getting-long-data */
+				return true;
+			}
+		}
+		else
+		{
+			setOdbcError (SQL_HANDLE_STMT, sqlStmt, "getLongData", false, errorKey);
+			SQLFreeHandle (SQL_HANDLE_STMT, sqlStmt);
+			return false;
+		}
+
+		iteration++;
+
+	} while (getDataRet != SQL_NO_DATA);
+
+	return true;
+}
+
+/**
+ * @brief Fetch the data that was queried by executing a SELECT statement
+ *
+ * @pre The @p sqlStmt must have been executed before calling this function with it as an argument.
+ * This function also converts the data retrieved from the datasource to a KeySet (keys + associated values and metadata)
+ *
+ * @param sqlStmt An executed SQL statement
+ * @param buffers A struct with the pre-defined output buffers that are used to retrieve that data of the queried columns
+ * 	See \ref struct columnData "struct columnData".
+ * @param[out] errorKey Used to store errors and warnings
+ *
+ * @returns The KeySet with the data retrieved from the SQL SELECT query.
+ * 	Make sure to ksDel() the returned KeySet.
+ * @retval NULL if no data was fetched or an error occurred (see @p errorKey for details)
+ *
+ * @see executeSqlStatement() for executing a prepared SQL statement
+ */
+static KeySet * fetchResults (SQLHSTMT sqlStmt, struct columnData * buffers, Key * errorKey)
+{
+	SQLRETURN ret;
+	KeySet * ksResult = NULL;
+
+	/* Needed for detecting if the current row contains a new key or just a further metakey for a previous key (outer join) */
+	char * prevKeyName = NULL;
+	Key * curKey = NULL;
+	Key * prevKey = NULL;
+
+	if (!sqlStmt || !buffers)
+	{
+		if (errorKey)
+		{
+			ELEKTRA_SET_INTERFACE_ERROR (
+				errorKey, "A NULL pointer was given as an argument to fetchResults() for 'sqlStmt' or 'columnData'");
+		}
+		return NULL;
+	}
+
+	while ((ret = SQLFetch (sqlStmt)) != SQL_NO_DATA)
+	{
+
+		if (!SQL_SUCCEEDED (ret))
+		{
+			setOdbcError (SQL_HANDLE_STMT, sqlStmt, "fetchResults", false, errorKey);
+			elektraFree (prevKeyName);
+
+			if (ksResult)
+			{
+				ksDel (ksResult);
+			}
+
+			SQLFreeHandle (SQL_HANDLE_STMT, sqlStmt);
+			return NULL;
+		}
+		else
```

Comment:
You can remove the else here to reduce nesting.

Reply:
Good catch! 👍🏻